### PR TITLE
Disable signing without a transaction tag

### PR DESCRIPTION
**Pull request by janezpodhostnik**
references: https://github.com/dapperlabs/flow-go/issues/5204

This PR disables transactions signed without a transaction tag being valid.

This PR is prepared for when we want to disable compatibility mode for signing without a tag. @psiemens

### CODE REVIEW

**Comment by ramtinms** [comment]
Do we know for how long we want to keep compatibility before this switch ?

File: fvm/fvm_test.go [hunk 1]
```diff
@@ -1848,7 +1848,8 @@ func TestSigningWithTags(t *testing.T) {
 
 				err = vm.Run(ctx, tx, view, programs)
 				require.NoError(t, err)
-				require.NoError(t, tx.Err)
+				require.Error(t, tx.Err)
+				require.Equal(t, errors.ErrCodeInvalidProposalSignatureError, tx.Err.(errors.Error).Code())
```
Comment (tarakby) [inline fvm/fvm_test.go:1852]: I think we can use `require.ErrorIs` here

**Review by tarakby — APPROVED** [review]
Looks good 👌

File: fvm/fvm_test.go (hunk 1 above)
Comment (janezpodhostnik) [reply fvm/fvm_test.go]: ErrrorIs and ErrorAs didn't work for me here. Not sure whats going on. I would probably need to define Is ad As on the error types.

I instead switched to IsType.

File: fvm/fvm_test.go [hunk 2]
```diff
@@ -1820,7 +1820,7 @@ func TestBlockContext_ExecuteTransaction_FailingTransactions(t *testing.T) {
 	)
 }
 func TestSigningWithTags(t *testing.T) {
-	t.Run("Signing Transactions without tag works (for now)", newVMTest().
+	t.Run("Signing Transactions without tag doesn't work", newVMTest().
```
Comment (m4ksio) [inline fvm/fvm_test.go:1823]: Do we also have tests with a tag, just for completeness?

Comment (janezpodhostnik) [reply fvm/fvm_test.go]: added

**Review by m4ksio — APPROVED** [review]

**Pull request merged**